### PR TITLE
IIs can initiate a call on the messages

### PR DESCRIPTION
**Pull request by gabrielsaruhashi**
## Description
This creates the foundational flow for starting a voice call. It successfully contacts the API, creates a meeting entity, and opens a voice call modal accordingly.

I included to-do notes outlining what's missing, but basically, we're missing the actual call connections. Hoping to use the connect call client for this. Will do this on a separate PR as this one was getting pretty hefty

https://user-images.githubusercontent.com/12942772/147421736-fc2f27d6-de92-4112-88a7-ec60f2c11e7d.mov

### CODE REVIEW

File: src/pages/InboxPage.tsx [hunk 1]
```diff
@@ -14,32 +14,36 @@ import * as Sentry from "@sentry/react";
...
-      onCompleted: ((data) => {
-        setCurrentInmateId(data?.currentInmate?.id);
-        if (data.currentInmate && !data.currentInmate.hasSentAnyMessages) {
-          setHasSentAnyMessages(false);
-          if (newUserIntroModalState === "neverOpened") {
-            setNewUserIntroModalState("open");
-          }
+  const [newUserIntroModalState, setNewUserIntroModalState] =
+    useState<ModalStates>("neverOpened");
+  const [firstMessageSentModalState, setFirstMessageSentModalState] =
```
Comment (gabrielsaruhashi) [inline src/pages/InboxPage.tsx:31]: prettier changes

File: src/components/Avatar/Facepile.tsx [hunk 2]
```diff
@@ -6,17 +6,17 @@ import { getFullName, getInitials } from "src/utils";
 import { getAvatarBackgroundColor } from "src/utils/UI";
 
 interface Props {
-  contacts: BasePersona[];
+  contacts: { firstName: string; lastName: string }[];
```
Comment (gabrielsaruhashi) [inline src/components/Avatar/Facepile.tsx:9]: these changes allow us to share these components between the graphql types and the hardcoded client types.

File: tailwind.config.js [hunk 3]
```diff
@@ -19,13 +19,14 @@ module.exports = {
...
+        200: "#B13E37",
+      },
     },
   },
   variants: {
     extend: {},
   },
   plugins: [],
-}
+  important: true,
```
Comment (gabrielsaruhashi) [inline tailwind.config.js:31]: overrides antd defaults

Comment (gabrielsaruhashi) [reply tailwind.config.js:31]: fixes several alignment problems

File: src/redux/modules/inPersonVisit.ts [hunk 4]
```diff
@@ -35,7 +35,10 @@ export const fetchInPersonVisits = createAsyncThunk(
     // convert to legacy data types
     return visits.map((visit) => ({
       id: visit.id,
-      meetingType: visit.meetingType,
+      meetingType:
```
Comment (gabrielsaruhashi) [inline src/redux/modules/inPersonVisit.ts:38]: neede bc of the new meeting type VoiceCall

File: src/hooks/useGraphqlError.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { ApolloError } from "@apollo/client";
+import { useEffect } from "react";
+import { showToast } from "src/utils";
+import * as Sentry from "@sentry/react";
+
+export default function useGraphqlError({
```
Comment (gabrielsaruhashi) [inline src/hooks/useGraphqlError.ts:6]: noticed we had been repeating this pattern everywhere that we called graphql using apollo
if we like it, I'll start refactor everywhere

File: src/pages/InboxPage.tsx [hunk 6]
```diff
@@ -64,88 +70,134 @@ const InboxPage: React.FC<Props> = () => {
...
-          id: selected?.id,
-        }),
-        fields: {
-          messages(existingRef) {
-            return { ...existingRef, unreadMessageCount: 0 }
-          }
-        }
-      }
-    );
+  useEffect(() => {
```
Comment (gabrielsaruhashi) [inline src/pages/InboxPage.tsx:96]: prettier

File: src/pages/InboxPage.tsx (hunk 1 above)
Comment (gabrielsaruhashi) [reply src/pages/InboxPage.tsx:33]: a lot of the changes in this file are prettier related.
It's surprising bc I was so certain that I had enforced prettier as a pre-commit hook 🧐  but doesn't seem to be the case
Im using prettier on auto-save now so these changes were all made automatically
Just look at the changes related to the voice call flow

File: src/components/Avatar/Facepile.tsx [hunk 7]
```diff
@@ -29,4 +29,4 @@ const ContactAvatarGroup: React.FC<Props> = ({ contacts, size, className }) => {
   );
 };
 
-export default ContactAvatarGroup;
+export default Facepile;
```
Comment (wittejm) [inline src/components/Avatar/Facepile.tsx:32]: "Facepile" is a hilarious word.

**Review by wittejm — APPROVED** [review]

File: src/components/Inbox/InboxHeader.tsx [hunk 8]
```diff
@@ -1,31 +1,57 @@
...
+            <Button
+              icon={<PhoneOutlined />}
+              loading={loading}
+              onClick={async () => {
+                setLoading(true);
+                try {
+                  await onCall();
+                } catch (err) {
+                  Sentry.captureException(err);
+                }
```
Comment (cainlevy) [inline src/components/Inbox/InboxHeader.tsx:41]: can we move error handling into the `onCall` handler?

File: src/graphql/fragments/FacilityServicesFragment.graphql [hunk 9]
```diff
@@ -1,4 +1,5 @@
 fragment FacilityServicesFragment on Facility {
   id
   hasMessaging
+  maxVoiceCallDuration
```
Comment (cainlevy) [inline src/graphql/fragments/FacilityServicesFragment.graphql:4]: i'm confused why this shows in the diff. it should be part of the base branch.

File: src/hooks/useFacilityFeature.ts [hunk 10]
```diff
@@ -1,7 +1,7 @@
 import { useQuery } from "@apollo/client";
 import { GetFacilityServicesDocument } from "src/graphql/queries/GetFacilityServices.generated";
 
-const features = ["MESSAGING"] as const;
+const features = ["MESSAGING", "VOICE_CALL"] as const;
```
Comment (cainlevy) [inline src/hooks/useFacilityFeature.ts:4]: huh, this is also supposed to be in the base branch.

File: src/graphql/fragments/MeetingFragment.graphql [hunk 11]
```diff
@@ -0,0 +1,9 @@
+fragment MeetingFragment on Meeting {
+    id
+    interval {
+        startAt 
+        endAt
+        seconds
+    }
```
Comment (cainlevy) [inline src/graphql/fragments/MeetingFragment.graphql:7]: how is this data used for voice calls?

File: src/hooks/useGraphqlError.ts (hunk 5 above)
Comment (cainlevy) [reply src/hooks/useGraphqlError.ts:6]: what about:

- error boundaries
- apollo's [`onError`](https://www.apollographql.com/docs/react/data/mutations/#onerror)

File: src/pages/InboxPage.tsx [hunk 12]
```diff
@@ -64,88 +71,138 @@ const InboxPage: React.FC<Props> = () => {
...
+    });
   }, [selected, currentInmateId]);
 
+  const handleVoiceCallCreation = async (connection: TConnectionListItem) => {
+    // open modal for call
+    // TODO: figure out how we want to identify a kiosk automatically
+    // https://app.shortcut.com/ameelio/story/2046/figure-out-how-we-want-to-identify-a-kiosk-automatically
+    const kioskId = localStorage.getItem(LOCAL_STORAGE_KIOSK_ID_KEY);
+
+    if (!kioskId) throw new Error("Did not setup kiosk correctly");
```
Comment (cainlevy) [inline src/pages/InboxPage.tsx:131]: is this just a dev shortcut? if so, could you clarify here so we don't confuse it with an implementation choice?

File: src/pages/InboxPage.tsx (hunk 1 above)
Comment (cainlevy) [reply src/pages/InboxPage.tsx:34]: maybe we could fix prettier separately so it doesn't make this pull request as noisy?

**Review by cainlevy — APPROVED** [review]

File: src/components/Avatar/Facepile.tsx (hunk 7 above)
Comment (gabrielsaruhashi) [reply src/components/Avatar/Facepile.tsx:32]: hahah it's what we called it at Facebook!

**Pull request merged**